modules/camera/camera_android.cpp

```diff
@@ -93,23 +95,91 @@ CameraFeedAndroid::~CameraFeedAndroid() {
+		ACameraMetadata_free(metadata);
+		metadata = nullptr;
+	}
+
+	camera_status_t status = ACameraManager_getCameraCharacteristics(manager, camera_id.utf8().get_data(), &metadata);
+	if (status != ACAMERA_OK || metadata == nullptr) {
+		ERR_PRINT(vformat("Camera %s: Failed to refresh metadata (status: %d)", camera_id, status));
+		return;
+	}
+
+	ACameraMetadata_const_entry orientation_entry;
+	status = ACameraMetadata_getConstEntry(metadata, ACAMERA_SENSOR_ORIENTATION, &orientation_entry);
+	if (status == ACAMERA_OK) {
+		orientation = orientation_entry.data.i32[0];
+		print_verbose(vformat("Camera %s: Orientation updated to %d", camera_id, orientation));
+	} else {
+		ERR_PRINT(vformat("Camera %s: Failed to get sensor orientation after refresh (status: %d)", camera_id, status));
+	}
+
+	formats.clear();
+	_add_formats();
+
+	print_verbose(vformat("Camera %s: Metadata refreshed successfully", camera_id));
+}
+
 void CameraFeedAndroid::_set_rotation() {
-	int display_rotation = DisplayServerAndroid::get_singleton()->get_display_rotation();
-	// reverse rotation
-	switch (display_rotation) {
-		case 90:
-			display_rotation = 270;
-			break;
-		case 270:
-			display_rotation = 90;
-			break;
-		default:
-			break;
+	if (!metadata) {
+		print_verbose(vformat("Camera %s: metadata is null in _set_rotation, attempting refresh", camera_id));
+		refresh_camera_metadata();
 	}
 
-	int sign = position == CameraFeed::FEED_FRONT ? 1 : -1;
-	float imageRotation = (orientation - display_rotation * sign + 360) % 360;
-	transform.set_rotation(real_t(Math::deg_to_rad(imageRotation)));
+	float image_rotation = 0.0f;
+
+	if (metadata) {
+		CameraRotationParams params;
+		params.sensorOrientation = orientation;
+		params.cameraFacing = (position == CameraFeed::FEED_FRONT) ? CameraFacing::FRONT : CameraFacing::BACK;
+		params.displayRotation = get_app_orientation();
+
+		RotationResult result = calculate_rotation(params);
+		if (result.isValid) {
+			image_rotation = static_cast<float>(result.rotationAngle);
+		} else {
+			int display_rotation = DisplayServerAndroid::get_singleton()->get_display_rotation();
+			switch (display_rotation) {
+				case 90:
+					display_rotation = 270;
+					break;
+				case 270:
+					display_rotation = 90;
+					break;
+				default:
+					break;
+			}
+
+			int sign = position == CameraFeed::FEED_FRONT ? 1 : -1;
+			image_rotation = (orientation - display_rotation * sign + 360) % 360;
+		}
+	} else {
+		ERR_PRINT(vformat("Camera %s: Cannot update rotation, metadata unavailable after refresh, using fallback", camera_id));
+		int display_rotation = DisplayServerAndroid::get_singleton()->get_display_rotation();
+		switch (display_rotation) {
+			case 90:
+				display_rotation = 270;
+				break;
+			case 270:
+				display_rotation = 90;
+				break;
+			default:
+				break;
+		}
+
+		int sign = position == CameraFeed::FEED_FRONT ? 1 : -1;
+		image_rotation = (orientation - display_rotation * sign + 360) % 360;
+	}
```

Comment:
To avoid duplicating the fallback logic, this can be rewritten as:

```suggestion
    RotationResult result = { 0, false, false };

	if (metadata) {
		CameraRotationParams params;
		params.sensorOrientation = orientation;
		params.cameraFacing = (position == CameraFeed::FEED_FRONT) ? CameraFacing::FRONT : CameraFacing::BACK;
		params.displayRotation = get_app_orientation();

		result = calculate_rotation(params);
	} else {
		ERR_PRINT(vformat("Camera %s: Cannot update rotation, metadata unavailable after refresh, using fallback", camera_id));
    }
	
	if (result.isValid) {
		image_rotation = static_cast<float>(result.rotationAngle);
	} else {
		int display_rotation = DisplayServerAndroid::get_singleton()->get_display_rotation();
		switch (display_rotation) {
			case 90:
				display_rotation = 270;
				break;
			case 270:
				display_rotation = 90;
				break;
			default:
				break;
		}

		int sign = position == CameraFeed::FEED_FRONT ? 1 : -1;
		image_rotation = (orientation - display_rotation * sign + 360) % 360;
	}
```